crawl-ref/source/mon-death.cc

```diff
@@ -1039,6 +1040,22 @@ static void _infestation_create_scarab(monster* mons)
     infestation_death_fineff::schedule(mons->pos(), mons->name(DESC_THE));
 }
 
+static void _pharaoh_ant_bind_souls(monster *mons)
+{
+    simple_monster_message(*mons, " binds the souls of nearby monsters.");
+    for (monster_near_iterator mi(mons, LOS_NO_TRANS); mi; ++mi)
+    {
+        if (*mi == mons)
+            continue;
+        if (mons_can_bind_soul(mons, *mi))
+        {
+            mi->add_ench(
+                mon_enchant(ENCH_BOUND_SOUL, 0, mons,
+                            random_range(10, 30) * BASELINE_DELAY));
+        }
+    }
```

Comment:
Let's only print this message if something gets bound.

```suggestion
    bool bound = false;
    for (monster_near_iterator mi(mons, LOS_NO_TRANS); mi; ++mi)
    {
        if (*mi == mons || !mons_can_bind_soul(mons, *mi))
            continue;
        if (!bound)
            simple_monster_message(*mons, " binds the souls of nearby monsters.");
        bound = true;
        mi->add_ench(
            mon_enchant(ENCH_BOUND_SOUL, 0, mons,
                        random_range(10, 30) * BASELINE_DELAY));
    }
```